components/core/src/os/process/mod.rs

```diff
@@ -151,8 +154,8 @@ impl fmt::Display for Signal {
 /// Encapsulates logic for defining the default shutdown signal we
 /// send services, and handles translation from external types at the
 /// edges of our system.
-#[derive(Debug, Clone)]
-pub struct ShutdownSignal(Signal);
+#[derive(Deserialize, Debug, Clone, Copy)]
+pub struct ShutdownSignal(#[serde(with = "serde_string")] Signal);
```

Comment:
This is cool, but after doing some digging, I see that we could _also_ do something like this, using the [from](https://serde.rs/container-attrs.html#from) attribute:

```rust
#[derive(Deserialize, Debug, Clone, Copy)]
#[serde(from = "Signal")]
pub struct ShutdownSignal(Signal);
```

Using `serde_string` is good, too, so I bring it up mainly as a way of sharing something nifty I just uncovered :smile: As we get a bit more principled with how we're using Serde, though, I think it would be good to dig further into the crate to try and uncover ways like this to leverage existing code.

(I went on a search for other alternatives, because I find it a little amazing that there doesn't seem to be support built into Serde for automatically taking advantage of existing `Display` and `FromStr` implementations on a type to do serialization and deserialization.)

I really do like how you refactored the `serde_string` stuff to make it more ergonomic to use, though; that's a nice touch!

Reply:
I had not seen "from" that is cool. I found this open [issue](https://github.com/serde-rs/serde/issues/1316) about adding documentation for the method I used. Which makes me think that maybe it is their preferred method? There `serialize` and `deserialize` implementations are simpler than ours so we could maybe clean it up more.

Reply:
Good find! Yeah, that implementation looks nice (especially with the right trait bounds).

I like it 👍